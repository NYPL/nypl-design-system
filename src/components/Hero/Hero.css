@import "../../styles/base/_03-mixins.css";

.hero {
  background-color: var(--ui-gray-warm-xlight);
}

.hero--50-50 {
  .hero__content {
    @add-mixin wrapper;

    align-items: center;
    display: flex;
    flex-flow: column nowrap;
    padding: 0 var(--space) var(--space) var(--space);

    @media (--large-break) {
      flex-flow: row nowrap;
      padding: unset;
    }
  }

  .hero__image {
    margin-bottom: var(--space-s);

    @media (--large-break) {
      margin-bottom: unset;
      margin-right: var(--space-s);
      max-width: 50%;
    }
  }
}

.hero--primary {
  align-items: center;
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  min-height: 350px;

  @media (--medium-break) {
    flex-flow: row nowrap;
  }

  .hero__content {
    background-color: var(--ui-black);
    color: var(--ui-white);
    flex: 0 0 100%;
    padding: var(--space-xxl) var(--space-l);

    a {
      color: inherit;
      display: inline-block;
    }

    @media (--medium-break) {
      flex: 0 0 60%;
      max-width: 960px;
    }

    + .placeholder {
      flex: 0 1 200px;
      margin-left: var(--space-xl);
    }
  }

  .hero__heading,
  .hero__body-text {
    margin-bottom: 0;
  }
}

.hero--secondary {
  @add-mixin breakout;

  overflow-x: hidden;

  .hero__content {
    @add-mixin wrapper;

    display: flex;
    flex-flow: column nowrap;

    @media (--medium-break) {
      flex-flow: row wrap;
    }

    .hero__heading {
      background-color: var(--ui-black);
      color: var(--ui-white);
      flex: 1 1 100%;
      margin-bottom: 0;
      margin-top: 0;
      padding-bottom: var(--space-xxs);
      position: relative;
      z-index: 0;

      &::before {
        background-color: var(--ui-black);
        content: "";
        height: 100%;
        left: -2000px;
        position: absolute;
        width: 4000px;
        z-index: -1;
      }

      @media (--medium-break) {
        order: 1;
      }

      .nypl--books-and-more & {
        background-color: var(--section-books-and-more-primary);

        &::before {
          background-color: var(--section-books-and-more-primary);
        }
      }

      .nypl--locations & {
        background-color: var(--section-locations-primary);

        &::before {
          background-color: var(--section-locations-primary);
        }
      }

      .nypl--research & {
        background-color: var(--section-research-primary);

        &::before {
          background-color: var(--section-research-primary);
        }
      }

      .nypl--whats-on & {
        background-color: var(--section-whats-on-primary);

        &::before {
          background-color: var(--section-whats-on-primary);
        }
      }
    }

    .hero__body-text {
      margin-top: var(--space-xs);

      @media (--medium-break) {
        flex: 1 1 50%;
        margin-right: var(--space-xs);
        order: 2;
      }
    }

    .hero__image {
      flex: 1 1 100%;
      height: 150px;
      /* https://github.com/philipwalton/flexbugs/issues/41 */
      min-width: 0;
      object-fit: cover;
      width: 100%;

      @media (--medium-break) {
        flex: 0 0 250px;
        order: 3;
      }
    }
  }
}
