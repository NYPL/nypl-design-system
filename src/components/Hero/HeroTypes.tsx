export enum HeroTypes {
  Primary = "PRIMARY",
  Secondary = "SECONDARY",
  Tertiary = "TERTIARY",
  FiftyFifty = "FIFTYFIFTY",
}
