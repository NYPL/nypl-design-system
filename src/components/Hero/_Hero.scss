.hero {
  background-color: var(--ui-gray-warm-xlight);
}

.hero--50-50 {
  .hero__content {
    @include wrapper;

    align-items: center;
    display: flex;
    flex-flow: column nowrap;
    padding: 0 var(--space) var(--space) var(--space);

    @include breakpoint($breakpoint-large) {
      flex-flow: row nowrap;
      padding: unset;
    }
  }

  .hero__image {
    margin-bottom: var(--space-s);

    @include breakpoint($breakpoint-large) {
      margin-bottom: unset;
      margin-right: var(--space-s);
      max-width: 50%;
    }
  }
}

.hero--primary {
  align-items: center;
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  min-height: 350px;

  @include breakpoint($breakpoint-medium) {
    flex-flow: row nowrap;
  }

  .hero__content {
    background-color: var(--ui-black);
    color: var(--ui-white);
    flex: 0 0 100%;
    padding: var(--space-xxl) var(--space-l);

    a {
      color: inherit;
      display: inline-block;
    }

    @include breakpoint($breakpoint-medium) {
      flex: 0 0 60%;
      max-width: 960px;
    }

    + .placeholder {
      flex: 0 1 200px;
      margin-left: var(--space-xl);
    }
  }

  .hero__heading,
  .hero__body-text {
    @include space-stack-none;
  }
}

.hero--secondary {
  overflow-x: hidden;

  .hero__content {
    @include wrapper;

    display: flex;
    flex-flow: column nowrap;

    @include breakpoint($breakpoint-medium) {
      flex-flow: row wrap;
    }

    .hero__heading {
      @include space-stack-none;

      background-color: var(--ui-black);
      color: var(--ui-white);
      flex: 1 1 100%;
      margin-top: 0;
      padding-bottom: var(--space-xxs);
      position: relative;
      z-index: 0;

      &::before {
        background-color: var(--ui-black);
        content: "";
        height: 100%;
        left: -2000px;
        position: absolute;
        width: 4000px;
        z-index: -1;
      }

      @include breakpoint($breakpoint-medium) {
        order: 1;
      }

      .nypl--books-and-more & {
        background-color: var(--section-books-and-more-primary);

        &::before {
          background-color: var(--section-books-and-more-primary);
        }
      }

      .nypl--locations & {
        background-color: var(--section-locations-primary);

        &::before {
          background-color: var(--section-locations-primary);
        }
      }

      .nypl--research & {
        background-color: var(--section-research-primary);

        &::before {
          background-color: var(--section-research-primary);
        }
      }

      .nypl--whats-on & {
        background-color: var(--section-whats-on-primary);

        &::before {
          background-color: var(--section-whats-on-primary);
        }
      }
    }

    .hero__body-text {
      margin-top: var(--space-xs);

      @include breakpoint($breakpoint-medium) {
        @include space-inline-xs;

        flex: 1 1 50%;
        order: 2;
      }
    }

    .hero__image {
      flex: 1 1 100%;
      height: 150px;
      min-width: 0; // https://github.com/philipwalton/flexbugs/issues/41
      object-fit: cover;
      width: 100%;

      @include breakpoint($breakpoint-medium) {
        flex: 0 0 250px;
        order: 3;
      }
    }
  }
}

.hero--tertiary {
  background-color: var(--ui-gray-xdark);

  .hero__content {
    @include wrapper;

    color: var(--ui-white);
    display: flex;
    flex-flow: column nowrap;
    padding: var(--space-l) var(--space);

    /* stylelint-disable */
    .hero__heading {
      @include space-stack-s;

      &:last-child {
        @include space-stack-none;
      }
    }
    /* stylelint-enable */
    p {
      @include space-stack-none;
    }
  }
}
