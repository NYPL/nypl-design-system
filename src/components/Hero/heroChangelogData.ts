/** This data is used to populate the ComponentChangelogTable component.
 *
 * date: string (when adding new entry during development, set value as "Prerelease")
 * version: string (when adding new entry during development, set value as "Prerelease")
 * type: "Bug Fix" | "New Feature" | "Update";
 * affects: "Accessibility" | "Documentation" | "Functionality" | "Styles";
 * notes: array (will render as a bulleted list, add one array element for each list element)
 */
export const changelogData = [
  {
    date: "Prerelease",
    version: "Prerelease",
    type: "Bug Fix",
    affects: "Functionality",
    notes: [
      "Fixes an issue with `backgroundColor` and `foregroundColor` props not prioritizing the passed design token values for the `Hero` component.",
    ],
  },
  {
    date: "2023-9-28",
    version: "2.0.0",
    type: "Update",
    affects: "Styles",
    notes: [
      "Applied Typo2023 styles, including font size, font color, and text link patterns.",
    ],
  },
] as const;
