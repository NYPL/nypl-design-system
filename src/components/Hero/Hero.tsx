// OH-30 Header with Image Right
import * as React from "react";
import bem from "../../utils/bem";
import { HeroTypes } from "./HeroTypes";

export interface HeroProps {
  /** Content creators can modify the background color
   * when this component is used on Exhibition pages.
   */
  backgroundColor?: string;
  /** Image used for primary Hero types. Note, cannot
   * be used in conjunction with image.
   */
  backgroundImageSrc?: string;
  /** BlockName for use with BEM. See how to work with blockNames and BEM here: http://getbem.com/introduction/ */
  blockName?: string;
  /** ClassName that appears in addition to "hero" */
  className?: string;
  /** Content creators can modify the foreground color
   * when this component is used on Exhibition pages.
   */
  foregroundColor?: string;
  /** Required heading element. */
  heading?: JSX.Element;
  /** Can be Primary, secondary, tertiary, or 50/50. */
  heroType?: HeroTypes;
  /** Image used for secondary Hero types. Note, cannot
   * be used in conjunction with backgroundImageSrc.
   */
  image?: JSX.Element;
  /** Optional details area that contains location data. */
  locationDetails?: JSX.Element;
  /** Optional subheader that displays underneath the
   * required heading element.
   */
  subHeaderText?: JSX.Element;
}

export default function Hero(props: React.PropsWithChildren<HeroProps>) {
  const heroBaseClass = "hero";

  const {
    backgroundColor,
    backgroundImageSrc,
    blockName,
    className,
    foregroundColor,
    heading,
    heroType,
    image,
    locationDetails,
    subHeaderText,
  } = props;

  let heroModifiers;

  if (heroType === HeroTypes.Primary) {
    heroModifiers = ["primary"];
  } else if (heroType === HeroTypes.Secondary) {
    heroModifiers = ["secondary"];
  } else if (heroType === HeroTypes.Tertiary) {
    heroModifiers = ["tertiary"];
  } else if (heroType === HeroTypes.FiftyFifty) {
    heroModifiers = ["50-50"];
  }

  if (heroType === HeroTypes.Primary && !backgroundImageSrc) {
    throw new Error(`backgroundImageSrc required on PRIMARY heroTypes`);
  }

  if (backgroundImageSrc && image) {
    throw new Error(
      `Please only either backgroundImageSrc or image into Hero, got both`
    );
  }

  if (heroType !== HeroTypes.Primary && locationDetails) {
    throw new Error(`Please provide locationDetails only to PRIMARY heroTypes`);
  }

  const backgroundImageStyle = backgroundImageSrc
    ? { backgroundImage: "url(" + backgroundImageSrc + ")" }
    : { backgroundColor: backgroundColor };

  let contentBoxStyling = {};
  if (
    heroType === HeroTypes.Primary ||
    heroType === HeroTypes.Tertiary ||
    heroType === HeroTypes.FiftyFifty
  ) {
    if (foregroundColor && backgroundColor) {
      contentBoxStyling = {
        color: foregroundColor,
        backgroundColor: backgroundColor,
      };
    } else if (foregroundColor || backgroundColor) {
      const receivedColor = foregroundColor
        ? "foregroundColor"
        : "backgroundColor";
      throw new Error(
        `Please provide both foregroundColor and backgroundColor to Hero, only got ` +
          receivedColor
      );
    }
  } else {
    if (foregroundColor || backgroundColor) {
      throw new Error(
        `Received foregroundColor and/or backgroundColor, but these are only accepted on HeroTypes.Primary and HeroTypes.FiftyFifty`
      );
    }
  }

  return (
    <div
      className={bem(heroBaseClass, heroModifiers, blockName, [className])}
      data-responsive-background-image
      style={backgroundImageStyle}
    >
      <div
        className={bem("content", [], heroBaseClass)}
        style={contentBoxStyling}
      >
        {heading}

        {image}

        {subHeaderText}
      </div>

      {locationDetails}
    </div>
  );
}
