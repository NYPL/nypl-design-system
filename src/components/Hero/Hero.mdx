import { ArgTypes, Canvas, Description, Meta, Source } from "@storybook/blocks";

import ComponentChangelogTable from "../../utils/ComponentChangelogTable";
import * as HeroStories from "./Hero.stories";
import { changelogData } from "./heroChangelogData";
import Link from "../Link/Link";

<Meta of={HeroStories} />

# Hero

| Component Version | DS Version |
| ----------------- | ---------- |
| Added             | `0.2.0`    |
| Latest            | `3.0.0`    |

## Table of Contents

- {<Link href="#overview" target="_self">Overview</Link>}
- {<Link href="#component-props" target="_self">Component Props</Link>}
- {<Link href="#accessibility" target="_self">Accessibility</Link>}
- {<Link href="#active-variants" target="_self">Active Variants</Link>}
- {<Link href="#deprecated-variants" target="_self">Deprecated Variants</Link>}
- {<Link href="#color-variations-for-secondary-hero" target="_self">Color Variations for Secondary Hero</Link>}
- {<Link href="#changelog" target="_self">Changelog</Link>}

## Overview

<Description of={HeroStories} />

The `Hero` component is used to display a full width banner at the top of a page.
The `Hero` will contain a required `h1` page title and may also include optional
descriptive text and images.

There are currently three main hero types: `"primary"`, `"tertiary"`, and
`"campaign"`.

Two hero types have been deprecated: `"secondary"` and `"fiftyFifty"`. The
deprecated variants will be removed from the DS in a future release.

Under the `"secondary"` hero type, there are four additional subtypes:
`"secondaryBooksAndMore"`, `"secondaryLocations"`, `"secondaryResearch"`, and
`"secondaryWhatsOn"`.

For the variations that use image "alt" and "src" attributes, the `imageProps`
prop should be used:

<Source
  code={`
<Hero
  backgroundImageSrc="https://placekitten.com/g/2400/800"
  heroType="campaign"
  heading={<Heading level="h1" id="campaign-hero" text="Hero Campaign" />}
  imageProps={{
    alt: "Image example",
    src: "https://placekitten.com/800/400",
  }}
  subHeaderText="Campaign Hero Subheader Text"
/>
`}
  language="jsx"
/>

Note: the `imageProps` prop is not the same as the `backgroundImageSrc` prop
used for some variations. For a full list of what props to use for each variant,
check the <Link href="#active-variants" target="_self">Active Variants</Link>
section.

## Component Props

<Canvas of={HeroStories.WithControls} />

<ArgTypes of={HeroStories.WithControls} />

## Accessibility

The `Hero` component is commonly used as a banner below the page's header and
breadcrumbs and above the main content. This means that if the `Hero` displays
a heading element, it should be the first on the page and an `h1` element set
through `Heading`'s `level="h1"`.

## Active Variants

Each `Hero` variation can be rendered through the `heroType` prop.

### Primary

The `"primary"` hero type can be used with the `backgroundColor`, `backgroundImageSrc`,
`foregroundColor`, `heading`, `locationDetails`, and `subHeaderText` props. The
minimum props that should be used are `backgroundImageSrc` and `heading`.

<Source
  code={`
backgroundImageSrc="https://placekitten.com/1600/800"
heading={<Heading level="h1" id="primary-hero" text="Hero Primary" />}
`}
  language="jsx"
/>

<Canvas of={HeroStories.Primary} />

### Tertiary

The `"tertiary"` hero type can be used with the `backgroundColor`, `foregroundColor`,
`heading`, and `subHeaderText` props. The minimum prop to use is the `heading`
prop.

<Source
  code={`
heading={
  <Heading
    level="h1"
    id="tertiary-hero"
    text="Hero Tertiary with Sub-Header"
  />
}
subHeaderText={otherSubHeaderText}
`}
  language="jsx"
/>

<Canvas of={HeroStories.Tertiary} />

### Campaign

The `"campaign"` hero type can be used with the `backdropBackgroundColor`,
`backgroundColor`, `backgroundImageSrc`, `foregroundColor`, `heading`,
`imageProps`, and `subHeaderText` props. The minimum props to use are
`backgroundImageSrc`, `heading`, and `imageProps`.

This variant has the special use case of having a background and foreground. We
do understand that the naming can be confusing, but it helps to think of this
variant having two layers. The bottom is the backdrop and the top is the
foreground.

To modify the foreground, the `backgroundColor`, `foregroundColor`, `imageProps`
props should be used.

<Source
  code={`
backgroundImageSrc="https://placekitten.com/g/2400/800"
heading={
  <Heading level="h1" id="campaign-hero" text="Hero Campaign" />
}
imageProps={imageProps}
subHeaderText={otherSubHeaderText}
`}
  language="jsx"
/>

<Canvas of={HeroStories.Campaign} />

#### isDarkBackgroundImage

The `isDarkBackgroundImage` prop can be used in the `"campaign"` hero to toggle
the visual treatment of the background image. If true, the background image will
be converted to black & white and darkened to 60% black.

<Source
  code={`
backgroundImageSrc="https://placekitten.com/g/2400/800"
heading={
  <Heading level="h1" id="campaign-hero" text="Hero Campaign" />
}
imageProps={imageProps}
isDarkBackgroundImage
subHeaderText={otherSubHeaderText}
`}
  language="jsx"
/>

<Canvas of={HeroStories.CampaignDarkBackgroundImage} />

#### backdropBackgroundColor

The "backdrop" is what we call the background and that can be set through either
the `backdropBackgroundColor` or `backgroundImageSrc` props. When both are set,
the `backgroundImageSrc` will take precedence.

<Source
  code={`
backdropBackgroundColor="section.education.primary"
backgroundColor="ui.warning.primary"
foregroundColor="ui.typgraphy.heading"
heading={
  <Heading 
    color="ui.typgraphy.heading" 
    level="h1" 
    id="campaign-hero" 
    text="Hero Campaign" 
  />
}
imageProps={imageProps}
subHeaderText={otherSubHeaderText}
`}
  language="jsx"
/>

<Canvas of={HeroStories.CampaignBackgroundColors} />

## Fallback Campaign Image

<Source
  code={`
const FallBackExample = () => {
  // Or get image src from external service.
  const [imageSrc, setImageSrc] = useState("foobar.jpg");
  const fallbackImageSrc = "https://loremflickr.com/cache/resized/65535_51121136167_40f20af042_z_300_300_nofilter.jpg";
  // For some reason IIIF image as fallback does not work...
  // Perhaps the better approach is to store the image locally in the app.
  // This cannot be easily tested in Storybook.
  // const fallbackImageSrc = "https://iiif.nypl.org/iiif/2/105224/full/!900,900/0/default.jpg";

  return (
    <Hero
      backgroundImageSrc={imageSrc}
      heroType="campaign"
      heading={
        <Heading
          level="h1"
          id="campaign-hero-default-heading"
          text="Hero Campaign"
        />
      }
      imageProps={{
        alt: "Custom NYPL",
        src: imageSrc,
        fallbackSrc: fallbackImageSrc,
        onError: (_event) => {
          setImageSrc(fallbackImageSrc);
        }
      }}
      subHeaderText={otherSubHeaderText}
    />
  );
};
`}
  language="jsx"
/>

<Canvas of={HeroStories.FallbackCampaignImage} />

## Deprecated Variants

### Secondary

This `"secondary"` variant of the `Hero` component has been deprecated. This
variant will be removed from the DS in a future release.

The `"secondary"` hero type can be used with the `heading`, `imageProps`, and
`subHeaderText` props.

<Source
  code={`
heading={
  <Heading level="h1" id="secondary-hero" text="Hero Secondary" />
}
imageProps={imageProps}
subHeaderText={secondarySubHeaderText}
`}
  language="jsx"
/>

<Canvas of={HeroStories.Secondary} />

### FiftyFifty

This `"fiftyFifty"` variant of the `Hero` component has been deprecated. This
variant will be removed from the DS in a future release.

The `"fiftyFifty"` hero type can be used with the `backgroundColor`,
`foregroundColor`, `heading`, `imageProps`, and `subHeaderText` props. The
minimum props to use are `imageProps` and `subHeaderText`.

<Source
  code={`
imageProps={imageProps}
subHeaderText={otherSubHeaderText}
`}
  language="jsx"
/>

<Canvas of={HeroStories.FiftyFifty} />

## Color Variations for Secondary Hero

This `"secondary"` variant of the `Hero` component has been deprecated. This
variant and the associated color variations will be removed from the DS in a
future release.

The background color for the title bar in the `"secondary"` `Hero` changes based
on the `heroType` variants for the `"secondary"` main variant. There are four
subtypes: `"secondaryBooksAndMore"`, `"secondaryLocations"`, `"secondaryResearch"`,
and `"secondaryWhatsOn"`.

<Source
  code={`
// Example \`heroType\` "secondary" variant.
<Hero heroType="secondaryBooksAndMore" {...props} />
`}
  language="jsx"
/>

<Canvas of={HeroStories.ColorVariations} />

## Changelog

<ComponentChangelogTable changelogData={changelogData} />
