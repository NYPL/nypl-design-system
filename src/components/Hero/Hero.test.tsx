import { expect } from "chai";
import * as Enzyme from "enzyme";
import * as React from "react";

import Placeholder from "../Placeholder/Placeholder";
import Image from "../Image/Image";
import Heading from "../Heading/Heading";
import { HeroTypes } from "./HeroTypes";
import Hero from "./Hero";

describe("Hero Test", () => {
  it("Generates a Hero with a background image", () => {
    const wrapper = Enzyme.shallow(
      <Hero
        heroType={HeroTypes.Primary}
        heading={
          <Heading
            level={1}
            id={"1"}
            text={"Hero Primary"}
            blockName={"hero"}
          />
        }
        backgroundImageSrc="https://placeimg.com/1600/800/arch"
      ></Hero>
    );
    expect(wrapper.prop("style")).to.deep.equal({
      backgroundImage: "url(https://placeimg.com/1600/800/arch)",
    });
  });

  it("Generates a Hero with a foreground image", () => {
    const wrapper = Enzyme.shallow(
      <Hero
        heroType={HeroTypes.Secondary}
        heading={
          <Heading
            level={1}
            id={"1"}
            text={"Hero Secondary"}
            blockName={"hero"}
          />
        }
        image={
          <Image
            src="https://placeimg.com/800/400/arch"
            blockName={"hero"}
            alt={""}
          />
        }
      ></Hero>
    );
    expect(wrapper.find("Image").dive().find("img")).to.have.lengthOf(1);
  });

  it("On primary hero, background image is required", () => {
    expect(() =>
      Enzyme.mount(
        <Hero
          heroType={HeroTypes.Primary}
          heading={
            <Heading
              level={1}
              id={"1"}
              text={"Hero Primary"}
              blockName={"hero"}
            />
          }
          image={
            <Image
              src="https://placeimg.com/800/400/arch"
              blockName={"hero"}
              alt={""}
            />
          }
        ></Hero>
      )
    ).to.throw("backgroundImageSrc required on PRIMARY heroTypes");
  });

  it("Throws error if both backgroundImage and foregroundImage are passed", () => {
    expect(() =>
      Enzyme.mount(
        <Hero
          heroType={HeroTypes.Secondary}
          heading={
            <Heading
              level={1}
              id={"1"}
              text={"Hero Secondary"}
              blockName={"hero"}
            />
          }
          image={
            <Image
              src="https://placeimg.com/800/400/arch"
              blockName={"hero"}
              alt={""}
            />
          }
          backgroundImageSrc="https://placeimg.com/1600/800/arch"
        ></Hero>
      )
    ).to.throw(
      "Please only either backgroundImageSrc or image into Hero, got both"
    );
  });

  it("Throws error if locationDetails are based to non-primary hero types", () => {
    expect(() =>
      Enzyme.mount(
        <Hero
          heroType={HeroTypes.Secondary}
          heading={
            <Heading
              level={1}
              id={"1"}
              text={"Hero Secondary"}
              blockName={"hero"}
            />
          }
          image={
            <Image
              src="https://placeimg.com/800/400/arch"
              blockName={"hero"}
              alt={""}
            />
          }
          locationDetails={
            <Placeholder>
              {"Placeholder for locationDetails, which doesn't exist yet"}
            </Placeholder>
          }
        ></Hero>
      )
    ).to.throw("Please provide locationDetails only to PRIMARY heroTypes");
  });

  it("Throws error if only one var is passed between foregroundColor and backgroundColor", () => {
    expect(() =>
      Enzyme.mount(
        <Hero
          heroType={HeroTypes.Primary}
          heading={
            <Heading
              level={1}
              id={"1"}
              text={
                "Syncretic Vibrations: Exploring the Mosaic of Blackness through the Melville J. and Frances S.Herskovits Collection"
              }
              blockName={"hero"}
            />
          }
          locationDetails={
            <Placeholder>
              {"Placeholder for locationDetails, which doesn't exist yet"}
            </Placeholder>
          }
          foregroundColor="#ffffff"
          backgroundImageSrc="https://p24.f4.n0.cdn.getcloudapp.com/items/NQuDO4xO/index.jpeg?v=d49888fbe420dd2fd163adc2ad0cdac6"
        />
      )
    ).to.throw(
      "Please provide both foregroundColor and backgroundColor to Hero, only got foregroundColor"
    );
  });
});
