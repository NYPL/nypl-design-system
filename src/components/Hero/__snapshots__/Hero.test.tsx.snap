// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Renders the UI snapshot correctly 1`] = `
<div
  className="css-1xdhyk6"
  data-responsive-background-image={true}
  data-testid="hero"
  style={
    Object {
      "backgroundImage": "url(https://placeimg.com/1600/800/arch)",
    }
  }
>
  <div
    className="css-0"
    data-testid="hero-content"
    style={
      Object {
        "backgroundColor": undefined,
        "color": undefined,
      }
    }
  >
    <h1
      className="chakra-heading css-1xdhyk6"
      id="primary-hero"
    >
      Hero Primary
    </h1>
    <div
      className="css-0"
    >
      Example Subtitle
    </div>
  </div>
</div>
`;

exports[`Hero Renders the UI snapshot correctly 2`] = `
<div
  className="css-1xdhyk6"
  data-responsive-background-image={true}
  data-testid="hero"
  style={Object {}}
>
  <div
    className="css-0"
    data-testid="hero-content"
    style={Object {}}
  >
    <div
      className="css-0"
    >
      <img
        alt="Image example"
        className="css-0"
        src="https://placeimg.com/800/400/animals"
      />
    </div>
    <h1
      className="chakra-heading css-1xdhyk6"
      id="secondary-hero"
    >
      Hero Secondary
    </h1>
    <div
      className="css-0"
    >
      Explore our collection of hundreds of online resources and databases. Use our free online content to help with your research, whether it's finding a single article, tracing a family tree, learning a new language, or anything in between.
    </div>
  </div>
</div>
`;

exports[`Hero Renders the UI snapshot correctly 3`] = `
<div
  className="css-1xdhyk6"
  data-responsive-background-image={true}
  data-testid="hero"
  style={Object {}}
>
  <div
    className="css-0"
    data-testid="hero-content"
    style={Object {}}
  >
    <div
      className="css-0"
    >
      <img
        alt="Image example"
        className="css-0"
        src="https://placeimg.com/800/400/animals"
      />
    </div>
    <h1
      className="chakra-heading css-1xdhyk6"
      id="secondary-hero"
    >
      Hero Secondary Books and More
    </h1>
    <div
      className="css-0"
    >
      Explore our collection of hundreds of online resources and databases. Use our free online content to help with your research, whether it's finding a single article, tracing a family tree, learning a new language, or anything in between.
    </div>
  </div>
</div>
`;

exports[`Hero Renders the UI snapshot correctly 4`] = `
<div
  className="css-1xdhyk6"
  data-responsive-background-image={true}
  data-testid="hero"
  style={Object {}}
>
  <div
    className="css-0"
    data-testid="hero-content"
    style={Object {}}
  >
    <div
      className="css-0"
    >
      <img
        alt="Image example"
        className="css-0"
        src="https://placeimg.com/800/400/animals"
      />
    </div>
    <h1
      className="chakra-heading css-1xdhyk6"
      id="secondary-hero"
    >
      Hero Secondary Locations
    </h1>
    <div
      className="css-0"
    >
      Explore our collection of hundreds of online resources and databases. Use our free online content to help with your research, whether it's finding a single article, tracing a family tree, learning a new language, or anything in between.
    </div>
  </div>
</div>
`;

exports[`Hero Renders the UI snapshot correctly 5`] = `
<div
  className="css-1xdhyk6"
  data-responsive-background-image={true}
  data-testid="hero"
  style={Object {}}
>
  <div
    className="css-0"
    data-testid="hero-content"
    style={Object {}}
  >
    <div
      className="css-0"
    >
      <img
        alt="Image example"
        className="css-0"
        src="https://placeimg.com/800/400/animals"
      />
    </div>
    <h1
      className="chakra-heading css-1xdhyk6"
      id="secondary-hero"
    >
      Hero Secondary
    </h1>
    <div
      className="css-0"
    >
      Explore our collection of hundreds of online resources and databases. Use our free online content to help with your research, whether it's finding a single article, tracing a family tree, learning a new language, or anything in between.
    </div>
  </div>
</div>
`;

exports[`Hero Renders the UI snapshot correctly 6`] = `
<div
  className="css-1xdhyk6"
  data-responsive-background-image={true}
  data-testid="hero"
  style={Object {}}
>
  <div
    className="css-0"
    data-testid="hero-content"
    style={Object {}}
  >
    <div
      className="css-0"
    >
      <img
        alt="Image example"
        className="css-0"
        src="https://placeimg.com/800/400/animals"
      />
    </div>
    <h1
      className="chakra-heading css-1xdhyk6"
      id="secondary-hero"
    >
      Hero Secondary What's On
    </h1>
    <div
      className="css-0"
    >
      Explore our collection of hundreds of online resources and databases. Use our free online content to help with your research, whether it's finding a single article, tracing a family tree, learning a new language, or anything in between.
    </div>
  </div>
</div>
`;

exports[`Hero Renders the UI snapshot correctly 7`] = `
<div
  className="css-1xdhyk6"
  data-responsive-background-image={true}
  data-testid="hero"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  <div
    className="css-0"
    data-testid="hero-content"
    style={
      Object {
        "backgroundColor": undefined,
        "color": undefined,
      }
    }
  >
    <h1
      className="chakra-heading css-1xdhyk6"
      id="tertiary-hero"
    >
      Hero Tertiary
    </h1>
    <p>
      With 92 locations across the Bronx, Manhattan, and Staten Island, The New York Public Library is an essential part of neighborhoods across the city. Visit us today.
    </p>
  </div>
</div>
`;

exports[`Hero Renders the UI snapshot correctly 8`] = `
<div
  className="css-1xdhyk6"
  data-responsive-background-image={true}
  data-testid="hero"
  style={
    Object {
      "backgroundImage": "url(https://placeimg.com/2400/800/nature/grayscale)",
    }
  }
>
  <div
    className="css-0"
    data-testid="hero-content"
    style={
      Object {
        "backgroundColor": undefined,
        "color": undefined,
      }
    }
  >
    <div
      className="css-0"
    >
      <img
        alt="Image example"
        className="css-0"
        src="https://placeimg.com/800/400/animals"
      />
    </div>
    <div
      className="css-0"
    >
      <h1
        className="chakra-heading css-1xdhyk6"
        id="campaign-hero"
      >
        Hero Campaign
      </h1>
      With 92 locations across the Bronx, Manhattan, and Staten Island, The New York Public Library is an essential part of neighborhoods across the city. Visit us today.
    </div>
  </div>
</div>
`;

exports[`Hero Renders the UI snapshot correctly 9`] = `
<div
  className="css-1xdhyk6"
  data-responsive-background-image={true}
  data-testid="hero"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  <div
    className="css-0"
    data-testid="hero-content"
    style={
      Object {
        "backgroundColor": undefined,
        "color": undefined,
      }
    }
  >
    <div
      className="css-0"
    >
      <img
        alt="Image example"
        className="css-0"
        src="https://placeimg.com/800/400/animals"
      />
    </div>
    <div
      className="css-0"
    >
      With 92 locations across the Bronx, Manhattan, and Staten Island, The New York Public Library is an essential part of neighborhoods across the city. Visit us today.
    </div>
  </div>
</div>
`;

exports[`Hero Renders the UI snapshot correctly 10`] = `
<div
  className="css-16h1ce8"
  data-responsive-background-image={true}
  data-testid="hero"
  style={
    Object {
      "backgroundImage": "url(https://placeimg.com/1600/800/arch)",
    }
  }
>
  <div
    className="css-0"
    data-testid="hero-content"
    style={
      Object {
        "backgroundColor": undefined,
        "color": undefined,
      }
    }
  >
    <h1
      className="chakra-heading css-1xdhyk6"
      id="chakra"
    >
      Hero Primary
    </h1>
    <div
      className="css-0"
    >
      Example Subtitle
    </div>
  </div>
</div>
`;

exports[`Hero Renders the UI snapshot correctly 11`] = `
<div
  className="css-1xdhyk6"
  data-responsive-background-image={true}
  data-testid="props"
  style={
    Object {
      "backgroundImage": "url(https://placeimg.com/1600/800/arch)",
    }
  }
>
  <div
    className="css-0"
    data-testid="hero-content"
    style={
      Object {
        "backgroundColor": undefined,
        "color": undefined,
      }
    }
  >
    <h1
      className="chakra-heading css-1xdhyk6"
      id="props"
    >
      Hero Primary
    </h1>
    <div
      className="css-0"
    >
      Example Subtitle
    </div>
  </div>
</div>
`;
